### ISSUE TITLE
[connectivity_plus] always returns ConnectivityResult.mobile on Web

### ISSUE THREAD

**Issue by sgehrman** [fluttercommunity/plus_plugins#639]
If I turn off my wifi or ethernet, I get an event, but it's always ConnectivityResult.mobile.

on or off doesn't matter.
effectiveType is always 4g.

**Comment by atultiwari** [comment]
yes me too facing the same issue

**Comment by harriseldon** [comment]
This appears to be the way that dart:html returns type or effectiveType
I created this gist against the dart:html package directly.
https://dartpad.dev/59e206f757d7fe1beb77633038924da4

It only works in Blink-based browsers (Chrome, Vivaldi, Edge). I tried in Firefox and did not get a result.

It does not appear that any of the other properties of html.NetworkInformation would suffice. Is this a dart upstream bug?

**Comment by dan1229** [comment]
+1 this - I get mobile on web (running on my desktop) even if wifi is off. Firefox and Chrome.

**Comment by Thauanny** [comment]
I'm still having this problem, even though I'm on wifi or with other connections, has anyone found a solution or other available package that works?

**Comment by dan1229** [comment]
No updates on my end - even with Flutter 3 the stuff is still broken. Anyone else having more luck?

**Comment by scfoley** [comment]
Using the universal_html package as a workaround in web for now until bug is resolved. Thought maybe it might help some others facing same issue:

Future<bool> hasInternet() async {
    final ConnectivityResult connectivityResult = await Connectivity().checkConnectivity();

    bool hasInternet;

    if (kIsWeb) {
      var connection = html.window.navigator.onLine;

      if(connection != null){
        hasInternet = connection!;
      }
      else{
        hasInternet = false;
      }
    } else {
      if (connectivityResult == ConnectivityResult.mobile ||
          connectivityResult == ConnectivityResult.wifi) {
        hasInternet = true;
      }
      else{
        hasInternet = false;
      }
    }

    return hasInternet;
  }

**Comment by navaronbracke** [comment]
I second @harriseldon and @scfoley 

The problem indeed is that `NetworkInformation` does not give any _useful_ information when the browser is disconnected.

Example:

online (via wifi network)
```
NetworkInformation:
downlink: 1.75
downlinkMax: null
effectiveType: 4g
type: null
```

offline (turning off wifi)
```
NetworkInformation:
downlink: 10
downlinkMax: null
effectiveType: 4g // <- I would expect effective type to not return 4g here, it probably returns the cached value in the browser
type: null
```

Since the [NetworkInformation API](https://developer.mozilla.org/en-US/docs/Web/API/Network_Information_API) is still *experimental* the plugin should not use it yet.

Only when it becomes a stable API and it provides a reliable way to detect online / offline changed (like how the online/offline API does it) should it be used.

**Comment by miquelbeltran** [comment]
Should be fixed by #904
